azext_iot/monitor/property.py

```diff
@@ -0,0 +1,96 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import datetime
+import isodate
+import time
+
+from azext_iot.constants import (
+    CENTRAL_ENDPOINT,
+    DEVICETWIN_POLLING_INTERVAL_SEC,
+    DEVICETWIN_MONITOR_TIME_SEC,
+)
+from azext_iot.central.providers.devicetwin_provider import CentralDeviceTwinProvider
+from azext_iot.central.models.devicetwin import DeviceTwin, Property
+
+
+def start_property_monitor(
+    cmd,
+    device_id,
+    app_id,
+    central_dns_suffix=CENTRAL_ENDPOINT,
+    polling_interval_seconds=DEVICETWIN_POLLING_INTERVAL_SEC,
+):
+    prev_twin = None
+
+    device_twin_provider = CentralDeviceTwinProvider(
+        cmd=cmd, app_id=app_id, device_id=device_id
+    )
+
+    while True:
+
+        raw_twin = device_twin_provider.get_device_twin(
+            central_dns_suffix=central_dns_suffix
+        )
+
+        twin = DeviceTwin(raw_twin)
+        if prev_twin:
+            change_d = compare_properties(
+                prev_twin.desired_property, twin.desired_property
+            )
+            change_r = compare_properties(
+                prev_twin.reported_property, twin.reported_property
+            )
+
+            if change_d:
+                print("Changes in desired properties:")
```

Comment:
Is the usage of this functionality specifically for the inner dev loop? 

Consider that it is easier to parse standard payload formats (like json, yaml etc) - what if someone wanted to build scripts around this functionality for a particular use case. In this case they would have to parse out/handle the basic string texts everytime there is a change.  If the change notifications were for example entirely json it would be easier to consume.

Reply:
The first iteration is intended for inner dev-loop only. We probably won't change this right now since I think our plan is to use it for customer evaluation/gathering feedback.

Happy to revisit the output style later if we are seeing this used as part of bigger workflows

#wontfix